src/sinks/axiom.rs

```diff
@@ -0,0 +1,408 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    config::{
+        log_schema, AcknowledgementsConfig, GenerateConfig, Input, SinkConfig, SinkContext,
+        SinkDescription,
+    },
+    sinks::{
+        elasticsearch::{ElasticsearchAuth, ElasticsearchConfig},
+        util::{http::RequestConfig, Compression},
+        Healthcheck, VectorSink,
+    },
+    tls::TlsConfig,
+};
+
+static CLOUD_URL: &str = "https://cloud.axiom.co";
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+pub(self) struct AxiomConfig {
+    url: Option<String>,
+    org_id: Option<String>,
+    token: String,
+    dataset: String,
+
+    #[serde(default)]
+    request: RequestConfig,
+    #[serde(default)]
+    compression: Compression,
+    tls: Option<TlsConfig>,
+    #[serde(
+        default,
+        deserialize_with = "crate::serde::bool_or_struct",
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    acknowledgements: AcknowledgementsConfig,
+}
+
+inventory::submit! {
+    SinkDescription::new::<AxiomConfig>("axiom")
+}
+
+impl GenerateConfig for AxiomConfig {
+    fn generate_config() -> toml::Value {
+        toml::from_str(
+            r#"token = "${AXIOM_TOKEN}"
+            dataset = "my-dataset"
+            url = "${AXIOM_URL}"
+            org_id = "${AXIOM_ORG_ID}""#,
+        )
+        .unwrap()
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "axiom")]
+impl SinkConfig for AxiomConfig {
+    async fn build(&self, cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
+        let mut request = self.request.clone();
+        request.headers.insert(
+            "X-Axiom-Org-Id".to_string(),
+            self.org_id.clone().unwrap_or_default(),
+        );
+        let mut query = HashMap::with_capacity(1);
+        query.insert(
+            "timestamp-field".to_string(),
+            log_schema().timestamp_key().to_string(),
+        );
+
+        // Axiom has a custom high-performance database that can be ingested
+        // into using our HTTP endpoints, including one compatible with the
+        // Elasticsearch Bulk API.
+        // This configuration wraps the Elasticsearch config to minimize the
+        // amount of code.
+        let elasticsearch_config = ElasticsearchConfig {
+            endpoint: self.build_endpoint(),
+            compression: self.compression,
+            auth: Some(ElasticsearchAuth::Basic {
+                user: "axiom".to_string(),
+                password: self.token.clone(),
+            }),
+            query: Some(query),
+            tls: self.tls.clone(),
+            request,
+            ..Default::default()
+        };
+
+        elasticsearch_config.build(cx).await
+    }
+
+    fn input(&self) -> Input {
+        Input::log()
```

Comment:
```suggestion
        Input::new(DataType::Metric | DataType::Log)
```

I think this needs an import as well though